python/in_toto_attestation/v1/resource_descriptor.py

```diff
@@ -0,0 +1,28 @@
+#  Wrapper class for in-toto attestation ResourceDescriptor protos.
+
+import in_toto_attestation.v1.resource_descriptor_pb2 as rdpb
+
+class ResourceDescriptor:
+    def __init__(self, name='', uri='', digest={}, content=bytes(), download_location='', media_type='', annotations={}):
```

Comment:
Why do we have default arguments which are empty values here? If these fields are required, the API user should supply them. If they are not required, we should not be providing faux values.

Reply:
Most of these fields are optional, but we still want to give API users the ability to set them through the constructor, hence the keyword arguments. We could use the more generic `*kwargs` parameter, though the onus would be on the API user to know which are acceptable `kwargs` to pass, and would make the constructor more complex too.

The only required fields are _one of_ name, uri, or digest. Unless there's something I'm fundamentally misunderstanding about how to code in Python, if we make these three fields positional arguments, they'll always be required, even if only one is needed, making the API user to always pass in a dummy value themselves.

So, how do you suggest API users set optional fields in the protobuf? The default protobuf constructor will set all fields to an empty/dummy value, so the default values passed in through the function arguments don't really change the contents of optional fields.

Reply:
So the options are:
* no defaults, make the user set an empty value / None for the ones they don't want to set
* defaults for all, user overrides the ones they want to set
* kwargs

I think each one has some pain points, but I do think it should be between option 1 and 2. Defaults for all looks strange but it might be the easiest for the user?